### PR TITLE
Jetty 10 - Improve test reliability

### PR DESCRIPTION
**Pull request by joakime**
Some more test reliability fixes

### CODE REVIEW

File: jetty-http2/http2-client/src/test/java/org/eclipse/jetty/http2/client/ConnectTimeoutTest.java [hunk 1]
```diff
@@ -50,6 +51,13 @@ public void testConnectTimeout() throws Exception
             @Override
             public void failed(Throwable x)
             {
+                if (x instanceof NoRouteToHostException)
+                {
+                    // Unable to connect to destination is not a failure.
+                    latch.countDown();
+                    return;
+                }
```
Comment (sbordet) [inline jetty-http2/http2-client/src/test/java/org/eclipse/jetty/http2/client/ConnectTimeoutTest.java]: @joakime I don't know what to say about this. 

We have already called `assumeConnectTimeout()` which verified that connecting to that address _indeed_ throws a `SocketTimeoutException`.

Few lines later we do it via `client.connect()` and we get a different exception?
We should really have `if (docker.screwsUpNetworking())`?

**Review by sbordet — CHANGES_REQUESTED** [review]

File: jetty-http2/http2-client/src/test/java/org/eclipse/jetty/http2/client/ConnectTimeoutTest.java (hunk 1 above)
Comment (joakime) [reply jetty-http2/http2-client/src/test/java/org/eclipse/jetty/http2/client/ConnectTimeoutTest.java]: Let me try a non-routable, local only, IP instead.

Comment (joakime) [reply jetty-http2/http2-client/src/test/java/org/eclipse/jetty/http2/client/ConnectTimeoutTest.java]: The 10.* network is how docker itself communicates among containers.

Comment (joakime) [reply jetty-http2/http2-client/src/test/java/org/eclipse/jetty/http2/client/ConnectTimeoutTest.java]: The original test is flaky, even on my local machine (no docker) this test will fail about 15% of the time.
For various reasons: "No Route to Host", "Host unreachable".
I'll use an IANA recommended host instead.

**Review by sbordet — APPROVED** [review]

**Pull request merged**